src/transactions.rs

```diff
@@ -311,18 +362,25 @@ impl<L: Ledger + Serialize + DeserializeOwned> Transactions<L> {
         hash: &TransactionHash<L>,
     ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
         let txn = self.with_hash(hash)?;
-        Ok(TransactionEditor::new(&mut self.store, txn))
+        Ok(TransactionEditor::new(self, txn))
     }
 
-    /// Get all the transactions timing out at the provided time.  There is no mutable way to get this
-    /// Collection because we should only be edditing one thing at a time
-    pub fn with_timeout(&self, timeout: u64) -> Result<Vec<Transaction<L>>, KeystoreError<L>> {
+    /// Get all the transactions timing out at the provided time.  
+    pub fn with_timeout(
+        &self,
+        timeout: u64,
+    ) -> Result<impl Iterator<Item = Transaction<L>> + '_, KeystoreError<L>> {
         let uids = self
             .expiring_txns
             .get(&timeout)
             .ok_or(KeyValueStoreError::KeyNotFound)?;
 
-        Ok(uids.iter().map(|uid| self.get(uid).unwrap()).collect())
+        Ok(uids.iter().map(move |uid| self.get(uid).unwrap()))
+    }
+
+    /// Remove a transaction from the pending index when it is known to have timed out
+    pub async fn remove_pending_txns(&mut self, timeout: u64) {
+        self.expiring_txns.remove(&timeout);
```

Comment:
Agree on the rename and yeah I was leaving this to controller layer because I wasn't sure we'd always only remove transactions from this index if they were rejected but it certainly seems like the case.

Reply:
Oh, wait, just realized -- we have to change something persistent here or else this change to `expiring_txns` will get undone the next time we load from disk. Since `expiring_txns` is populated based on timeout, I believe we need to set `timeout` to `None` for each transaction we remove here. Given that, I would maybe lean slightly towards also setting `status` to `Rejected`, because setting `timeout` to `None` without updating status seems like asking for inconsistencies.

Reply:
Yeah I agree if we are editing the transaction anyway we should set it's status at the same time.